cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -222,15 +222,15 @@ void SparsePoseGraph::ComputeConstraintsForScan(
   const mapping::SubmapId matching_id = submap_ids.front();
   const auto& constant_data = trajectory_nodes_.at(node_id).constant_data;
   const transform::Rigid2d pose = transform::Project2D(
-      constant_data->initial_pose *
+      constant_data->local_pose *
       transform::Rigid3d::Rotation(constant_data->gravity_alignment.inverse()));
-  const transform::Rigid2d optimized_pose =
+  const transform::Rigid2d global_pose =
       optimization_problem_.submap_data().at(matching_id).pose *
       sparse_pose_graph::ComputeSubmapPose(*insertion_submaps.front())
```

Comment:
We should probably also rename `ComputeSubmapPose()`. Maybe `ComputeLocalSubmapPose()`?